### PR TITLE
Move the Kafka Source to the knative-eventing namespace

### PR DESCRIPTION
**Pull request by matzew**
Signed-off-by: Matthias Wessendorf <mwessend@redhat.com>

Fixes #

## Proposed Changes

- move source to knative-eventing
- add old/deprecated controller replica
-remove config-xyz duplicates
- change the leader election cfg name for the source

**Release Note**

```release-note
The KafkaSource will be installed in the knative-eventing namespace, and the old controller in knative-sources is scalled to 0
```

**Docs**

### CODE REVIEW

**Review by pierDipi — COMMENTED** [review]
I like it, we need to make sure that the Kafka source is installable without installing eventing though.

File: config/source/configmaps/leader-election.yaml
Comment (antoineco) [inline config/source/configmaps/leader-election.yaml:18]: This will leave duplicates upon the next deployment, do we want to introduce that here?

Comment (matzew) [reply config/source/configmaps/leader-election.yaml:18]: Good question, @antoineco

**Comment by matzew** [comment]
@antoineco @pierDipi @lionelvillard I updated this PR


For now we will have `config-leader-election-kafka` CMs, however, I feel this is wrong, as expressed here:
https://github.com/knative-sandbox/eventing-kafka/pull/231

Also related:
https://github.com/knative/eventing/issues/4593

**Comment by antoineco** [comment]
So what's the plan, merge this, hold #231, then un-hold #231 after this is merged?

File: config/source/configmaps/leader-election.yaml
Comment (antoineco) [reply config/source/configmaps/leader-election.yaml]: I think we should avoid renaming things in this PR if they are unrelated to the original intention which is "Move the Kafka Source to the knative-eventing namespace".

**Comment by matzew** [comment]
@antoineco 

> I  think we should avoid renaming things in this PR if they are unrelated to the original intention which is "Move the Kafka Source to the knative-eventing namespace"

Sure, I updated earlier today, see: https://github.com/knative-sandbox/eventing-kafka/pull/224/files 

> So what's the plan, merge this, hold #231, then un-hold #231 after this is merged?

I'd think so - the #231 is already on hold ;-)

**Review by antoineco — APPROVED** [review]
/lgtm

**Pull request merged**
